File: 프로그래머스/1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr, int divisor) {
        StringBuilder sb = new StringBuilder();

        ArrayList<Integer> list = new ArrayList<>();

        for (int n : arr) {
            if (n % divisor == 0) {
                sb.append(n);
                list.add(n);
            }
        }
        if (list.size() == 0) {
            list.add(-1);
        }

        int[] answer = list.stream()
                .mapToInt(Integer::intValue)
                .toArray();
        Arrays.sort(answer);
        return answer;
    }
}

File: 프로그래머스/1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.kt
class Solution {
    fun solution(arr: IntArray, divisor: Int): IntArray {

        var arr1 = mutableListOf<Int>()
        for (i in arr){
            if (i % divisor==0){
                arr1.add(i)
            }
        }
        if (arr1.size== 0){
            arr1.add(-1)
        }
        arr1.sort()

        return arr1.toIntArray()
    }
}